README.md

```diff
@@ -15,23 +12,53 @@ Export BRANCH_ROLE as `staging`, `live`, or `custom`. Then run:
 curl -s https://raw.githubusercontent.com/rancher/charts-build-scripts/master/init.sh > /dev/null | sh
 ```
 
-## Building
+### Building
 
 `make`
 
-## Running
+### Running
 
 `./bin/charts-build-scripts`
 
-## Validation command
+### Validation command
 
 For more information on the validation command, please see [`docs/validate.md`](docs/validate.md).
 
-## Debugging
+### Debugging
 
 For more information on how to debug this project, please see [`docs/debugging.md`](docs/debugging.md).
 
+
+## Developing `charts-build-scripts`
+
+### How to Run
+
+```
+go run main.go
+```
+
+### How to Run Unit Tests
+
+```
+go test ./...
+```
+
+### How to Lint
+
+```
+golangci-lint run ./...
+```
+
+### How to Release
+
+Releases are done via a github action that uses [`goreleaser`] (https://goreleaser.com/).
+In order to release, simply tag the commit that you want the release
+to be based off of with a tag that is in semver format. `goreleaser`
+takes care of the rest.
+
```

Comment:
Could you provide a more detailed on the step by step on doing this? 
Imagine someone that never used `gorelease` before and also doesn't know the rules of versioning on this repo.

Reply:
Done. Let me know if I can improve any part of it.